src/cow-react/common/hooks/useSafeMemo.ts

```diff
@@ -0,0 +1,25 @@
+import { useMemo } from 'react'
+import { CurrencyAmount, NativeCurrency, Token } from '@uniswap/sdk-core'
+
+/**
+ * UseMemo effectively (by values) compare only primitive types and compare objects by links
+ * To get the best performance we need process objects changes manually
+ */
+export function useSafeMemo<T>(deps: unknown[] | null, memoCall: () => T): T {
```

Comment:
As mentioned on slack, use same param order as regular `useMemo`

Actually, why not just use (not sure the right syntax) `...params[]: Params<typeof useMemo>` or something like that to use the exact interface from `useMemo`